nipype/interfaces/dtitk/registration.py

```diff
@@ -1,394 +1,419 @@
+                         xor=['transform'], argstr='-euler %g %g %g')
+    deformation = traits.Tuple((1, 1, 1, 0, 0, 0), desc='(xx,yy,zz,xy,yz,xz)',
+                               xor=['transform'],
+                               argstr='-deformation %g %g %g %g %g %g')
 
 
 class affSymTensor3DVolOutputSpec(TraitedSpec):
-    out_file = traits.File(desc='cheese', exists=True)
+    out_file = File(exists=True)
 
 
 class affSymTensor3DVolTask(CommandLineDtitk):
     """
     Applies affine transform to a tensor volume
 
-                Example
-                -------
+    Example
+    -------
 
-                >>> import nipype.interfaces.dtitk as dtitk
-                >>> node = dtitk.affSymTensor3DVolTask()
-                >>> node.inputs.in_tensor = 'diffusion.nii'
-                >>> node.inputs.in_xfm = 'ants_Affine.txt'
-                >>> node.run() # doctest: +SKIP
-                """
+    >>> import nipype.interfaces.dtitk as dtitk
+    >>> node = dtitk.affSymTensor3DVolTask()
+    >>> node.inputs.in_file = 'im1.nii'
+    >>> node.inputs.transform = 'im2.nii'
+    >>> node.cmdline
+    'affineSymTensor3DVolume -in im1.nii -interp LEI -out im1_affxfmd.nii -reorient PPD -trans im2.nii'
+    >>> node.run() # doctest: +SKIP
+    """
     input_spec = affSymTensor3DVolInputSpec
     output_spec = affSymTensor3DVolOutputSpec
     _cmd = 'affineSymTensor3DVolume'
 
-    def _list_outputs(self):
-        outputs = self.output_spec().get()
-        outputs['out_file'] = os.path.abspath(self.inputs.out_file)
-        return outputs
-
 
 class affScalarVolInputSpec(CommandLineInputSpec):
-    in_volume = traits.Str(desc='moving volume', exists=True, mandatory=False,
-                           position=0, argstr="-in %s")
-    in_xfm = traits.Str(desc='transform to apply', exists=True,
-                        mandatory=False,
-                        position=1, argstr="-trans %s")
-    in_target = traits.Str(desc='', position=2, argstr="-target %s")
-    out_file = traits.Str(desc='', mandatory=False, position=3,
-                          argstr="-out %s", name_source="in_volume",
-                          name_template="%s_affxfmd.nii.gz")
+    in_file = File(desc='moving scalar volume', exists=True,
+                   argstr="-in %s", mandatory=True)
+    out_file = traits.Str(desc='output filename', exists=True,
+                          argstr="-out %s", name_source="in_file",
+                          name_template="%s_affxfmd", keep_extension=True)
+    transform = File(exists=True, argstr="-trans %s",
+                     xor=['target', 'translation', 'euler', 'deformation'], desc='transform to apply: specify an input transformation  file; parameters input will be ignored',)
+    interpolation = traits.Enum(0, 1, usedefault=True,
+                                argstr="-interp %s",
+                                desc='0=trilinear (def); 1=nearest neighbor')
```

Comment:
Before this goes into a release, you could consider making this:

```Python
interpolation = traits.Enum('trilinear', 'nearest', usedefault=True,
                            argstr='-interp %s',
                            desc='use trilinear or nearest neighbor interpolation')
```

And then, below:

```Python
class affScalarVolTask:
    ...
    def _format_arg(self, name, spec, value):
        if name == 'interpolation':
            value = {'trilinear': 0, 'nearest': 1}[value]
        super(affScalarVolTask, self)._format_arg(name, spec, value)
```

This is one way we can make interfaces a little more user friendly. Your call whether it's worth it. Your `desc` makes it clear, as is.